config/eip.tf

```diff
@@ -1,13 +1,15 @@
 resource "aws_eip" "cluster_loadbalancer_eip" {
+  count = var.enable_eip ? 1 : 0
+
   vpc              = true
   public_ipv4_pool = "amazon"
   tags             = merge(tomap({ "Name" : "${var.eks_cluster_name}-loadbalancer-eip" }), var.common_tags)
 }
 
 output "radar_base_eip_allocation_id" {
-  value = aws_eip.cluster_loadbalancer_eip.allocation_id
+  value = aws_eip.cluster_loadbalancer_eip != [] ? aws_eip.cluster_loadbalancer_eip[0].allocation_id : ""
```

Comment:
Is `value = var.enable_eip? aws_eip.cluster_loadbalancer_eip[0].allocation_id : null` more explicit than checking the empty list? Also, `null` drops the output if the resource is disabled so no outputs will be created with empty-string values. Similar comments apply to other places.

Reply:
I get this error when trying to check the condition:

```
│ Error: Incorrect condition type
│ 
│   on eip.tf line 10, in output "radar_base_eip_allocation_id":
│   10:   value = aws_eip.cluster_loadbalancer_eip ? aws_eip.cluster_loadbalancer_eip[0].allocation_id : ""
│     ├────────────────
│     │ aws_eip.cluster_loadbalancer_eip is empty tuple
│ 
│ The condition expression must be of type bool.
```

I'll change the values from `""` to `null`

Reply:
I meant checking `var.enable_eip` rather than `aws_eip.cluster_loadbalancer_eip`?

Reply:
I see, I get this error message:
```
╷
│ Error: Invalid index
│ 
│   on s3.tf line 142, in output "radar_base_s3_output_bucket_name":
│  142:   value = var.enable_s3 != [] ? aws_s3_bucket.intermediate_output_storage[0].bucket : null
│     ├────────────────
│     │ aws_s3_bucket.intermediate_output_storage is empty tuple
│ 
│ The given key does not identify an element in this collection value: the collection has no elements.
╵
```

Reply:
You have declared `var.enable_s3` in variables.tf as a boolean rather than list.

Reply:
Ah sorry, I was multitasking and didn't pay attention